evmjit/libevmjit-cpp/JitVM.cpp

```diff
@@ -40,36 +37,36 @@ bytesConstRef JitVM::execImpl(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _on
 	m_data.gasPrice		= static_cast<decltype(m_data.gasPrice)>(_ext.gasPrice);
 	m_data.callData 	= _ext.data.data();
 	m_data.callDataSize = _ext.data.size();
-	m_data.address      = eth2llvm(fromAddress(_ext.myAddress));
-	m_data.caller       = eth2llvm(fromAddress(_ext.caller));
-	m_data.origin       = eth2llvm(fromAddress(_ext.origin));
-	m_data.callValue    = eth2llvm(_ext.value);
-	m_data.coinBase     = eth2llvm(fromAddress(_ext.currentBlock.coinbaseAddress));
-	m_data.difficulty   = eth2llvm(_ext.currentBlock.difficulty);
-	m_data.gasLimit     = eth2llvm(_ext.currentBlock.gasLimit);
+	m_data.address      = eth2jit(fromAddress(_ext.myAddress));
+	m_data.caller       = eth2jit(fromAddress(_ext.caller));
+	m_data.origin       = eth2jit(fromAddress(_ext.origin));
+	m_data.callValue    = eth2jit(_ext.value);
+	m_data.coinBase     = eth2jit(fromAddress(_ext.currentBlock.coinbaseAddress));
+	m_data.difficulty   = eth2jit(_ext.currentBlock.difficulty);
+	m_data.gasLimit     = eth2jit(_ext.currentBlock.gasLimit);
 	m_data.number 		= static_cast<decltype(m_data.number)>(_ext.currentBlock.number);
 	m_data.timestamp 	= static_cast<decltype(m_data.timestamp)>(_ext.currentBlock.timestamp);
 	m_data.code     	= _ext.code.data();
 	m_data.codeSize 	= _ext.code.size();
-	m_data.codeHash		= eth2llvm(_ext.codeHash);
+	m_data.codeHash		= eth2jit(_ext.codeHash);
 
-	auto env = reinterpret_cast<Env*>(&_ext);
-	auto exitCode = m_engine.run(&m_data, env);
+	m_context.init(m_data, reinterpret_cast<evmjit::Env*>(&_ext));
```

Comment:
Not pretty but safe.
Whatever pointer is passed here will be available in callback functions implemented by JIT's user. JIT do not have the notion of ExtVM etc. The Env* is just an opaque pointer as the Env is only declared. It just passes a pointer through. So it is safe as long the JIT's user cast the pointer back to the same type when processing a callback request. 
https://github.com/ethereum/cpp-ethereum/blob/develop/evmjit/libevmjit-cpp/Env.cpp#L22